### ISSUE TITLE
SettlementsClient.GetOpenBalance() throws an exception

### ISSUE THREAD

**Issue by project01-wim** [Viincenttt/MollieApi#197]
Calling the SettlementsClient.GetOpenBalance() method throws an exception. Details: 

Cannot deserialize the current JSON array (e.g. [1,2,3]) into type 'System.Collections.Generic.Dictionary`2[System.Int32,System.Collections.Generic.Dictionary`2[System.Int32,Mollie.Api.Models.Settlement.SettlementPeriod]]' because the type requires a JSON object (e.g. {"name":"value"}) to deserialize correctly.
To fix this error either change the JSON to a JSON object (e.g. {"name":"value"}) or change the deserialized type to an array or a type that implements a collection interface (e.g. ICollection, IList) like List<T> that can be deserialized from a JSON array. JsonArrayAttribute can also be added to the type to force it to deserialize from a JSON array.
Path 'periods', line 1, position 147.

**Comment by Viincenttt** [comment]
Hi Wim,

Thank you for reporting this issue. I'm not able to reproduce it, so I need a little bit of help. Could you provide me with the raw json response that Mollie returns when you make the call to `GetOpenBalance`? You can get the raw response if you set a breakpoint on line 79 of the `BaseMollieClient`. Please give me the value of the `resultContent` variable. 

Kind regards,
Vincent

**Comment by project01-wim** [comment]
Hi Vincent,

here you go: 

{"resource":"settlement","id":"open","createdAt":"2020-11-11T07:10:53+00:00","status":"open","amount":{"value":"0.00","currency":"EUR"},"periods":[],"_links":{"self":{"href":"https://api.mollie.com/v2/settlements/open","type":"application/hal+json"},"documentation":{"href":"https://docs.mollie.com/reference/v2/settlements-api/get-open-settlement","type":"text/html"}}}

=> Periods: []

**Comment by Viincenttt** [comment]
Hi Wim,

Thank you, I have managed to reproduce the issue. In the example documentation that Mollie provided, the `periods` property is a JSON object. However, in the response you provided, it is actually a JSON array. To solve this issue, I have created a custom `JsonConverter` that return a empty Periods property if the object returned is a empty array. If the regular response is returned by Mollie, the code behaves as it did before.

I've added a unit test to verify this behaviour. It would be great if you are able to test this fix and verify that it solves your issue. If it does, I'll try to release a new version of the NuGet package as soon as possible. 

Kind regards,
Vincent
